### ISSUE TITLE
Multiple relationships to the same table in "Table Metadata", only one is available as a "group by"

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#34159]
### Describe the bug

If you model more than one foreign key relationship to the same table in the Admin > "Table Metadata" screen then only one of these is available in the query builder for grouping by.

Both relationships are available in the filter options.

You can manually join to same table in the query builder and it's available in the group by.

### To Reproduce

1. In Admin > Table Metadata > Sample Database > Orders
2. For "PRODUCT_ID" change the target field for the Foreign key reference to "People  > ID"
3. Now you've got both USER_ID and PRODUCT_ID pointing to the People table
![image](https://github.com/metabase/metabase/assets/4504437/82d461c2-70cb-4fe1-992b-631b5f06c43e)

4. New question > Sample Database > Orders
5. Click the add a filter button and note that the Product link shows User fields, this is what we've modelled and is correct in this case.
![image](https://github.com/metabase/metabase/assets/4504437/130fe913-b149-44fd-b903-3576d906b24b)

6. Click the "Pick a column to group by" button, note that the "Product" link is missing. This is incorrect.
![image](https://github.com/metabase/metabase/assets/4504437/25d706cd-4ad1-4f6e-b4c1-b1e5739eb4fe)

You can not group by anything in the second link to the People table, in this case name Product.

This not just an oddity cause by the nonsense link of product to people, I see this in my live instance with real data that this used to work for.

### Expected behavior

I should be able to group by all linked table fields.

### Logs

No logs in the browser or server.

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-NZ",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/117.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "17.0.2+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "17.0.2",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "17.0.2+8",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "sqlserver",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.1"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.4"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-09-18",
      "tag": "v0.47.2",
      "branch": "release-x.47.x",
      "hash": "536c24c"
    },
    "settings": {
      "report-timezone": "NZ"
    }
  }
}
```

### Severity

It's stopping users from doing basic analysis that they used to be able to do.

### Additional context

There's a work around to manually join the tables in the query builder but this misses the benefit of being shared with all users.

Seems to be related to a handful of issues I've seen with multiple joins to the same table including #32499.

**Comment by metamben** [comment]
I thin this has been fixed by #33845.

**Comment by lbrdnk** [comment]
Fixed in #33845.
